src/main/java/backend/Logic.java

```diff
@@ -331,6 +331,74 @@ private void revertLocalLabelsReplace(TurboIssue modifiedIssue, List<String> ori
     }
 
     /**
+     * Handles the result of updating an issue's state on server. Current implementation includes
+     * reverting back to the original state locally if the server update failed.
+     * @param isUpdateSuccessful
+     * @param locallyModifiedIssue
+     * @param isOpenOriginally
+     * @return
+     */
+    private boolean handleIssueStateUpdateResult(boolean isUpdateSuccessful,
+                                                 Optional<TurboIssue> locallyModifiedIssue,
+                                                 boolean isOpenOriginally) {
+        if (!locallyModifiedIssue.isPresent()) {
+            logger.error("Unable to edit issue state locally");
+            return false;
+        }
+
+        if (isUpdateSuccessful) {
+            return true;
+        }
+
+        logger.error("Unable to update model on server");
+        revertLocalStateEdit(locallyModifiedIssue.get(), isOpenOriginally);
+        return false;
+    }
+
+    /**
+     * Set the state of the issue in the {@link Logic#models} corresponding to {@code modifiedIssue} to
+     * {@code isOpenOriginally} if the current state on the issue is assigned at the same time as {@code modifiedIssue}
+     * @param modifiedIssue
+     * @param isOpenOriginally
+     */
+    private void revertLocalStateEdit(TurboIssue modifiedIssue, boolean isOpenOriginally) {
+        TurboIssue currentIssue = getIssue(modifiedIssue.getRepoId(), modifiedIssue.getId()).orElse(modifiedIssue);
+        LocalDateTime originalStateModifiedAt = modifiedIssue.getStateLastModifiedAt();
+        LocalDateTime currentStateModifiedAt = currentIssue.getStateLastModifiedAt();
+        boolean isCurrentStateModifiedOriginalState = originalStateModifiedAt.isEqual(currentStateModifiedAt);
+
+        if (isCurrentStateModifiedOriginalState) {
+            logger.info("Reverting state for issue " + currentIssue);
+            models.editIssueState(currentIssue.getRepoId(), currentIssue.getId(), isOpenOriginally);
+            refreshUI();
+        }
+    }
```

Comment:
This is a general comment. Use nested blocks to handle abnormal branches. The normal flow of the method should be in the un-nested code.